Description: fix use after free in qlocale_unix.cpp
 This patch avoids the use of the returned variable when the destructor
 gets called.
Author: Julien Cristau <julien.cristau@logilab.fr>
Bug-Debian: http://bugs.debian.org/681476
Origin: <vendor|upstream|other>, <url of original patch>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: 2012-07-20

--- qt4-x11-4.8.2.orig/src/corelib/tools/qlocale_unix.cpp
+++ qt4-x11-4.8.2/src/corelib/tools/qlocale_unix.cpp
@@ -52,7 +52,7 @@
 
 QT_BEGIN_NAMESPACE
 
-static const char *getSystemLocale()
+static QByteArray *getSystemLocale()
 {
 #if defined(Q_OS_QNX)
     static char buff[257];
